### PR TITLE
New Sass variable to change vertical rule width

### LINKED ISSUE #38856: Vertical rule not the correct border width
### Prerequisites

- [X] I have [searched](https://github.com/twbs/bootstrap/issues?utf8=%E2%9C%93&q=is%3Aissue) for duplicate or closed issues
- [X] I have [validated](https://html5.validator.nu/) any HTML to avoid common problems
- [X] I have read the [contributing guidelines](https://github.com/twbs/bootstrap/blob/main/.github/CONTRIBUTING.md)

### Describe the issue

When changing the `$border-width` the horizonal ruler changes. However, the vertical ruler does not because it's hardcoded to use `1px`:
```scss
// helpers/_vr.scss
.vr {
  /* .. */
  width: 1px;
  /* .. */
}
```

Could it maybe be changed to use something like the existing `$hr-border-width`?

### Reduced test cases

https://codepen.io/simo026q-the-reactor/pen/dyQRpXY

### What operating system(s) are you seeing the problem on?

Windows

### What browser(s) are you seeing the problem on?

Microsoft Edge

### What version of Bootstrap are you using?

v5.3.0

### PR DESCRIPTION
**Pull request by julien-deramond**
### Description

This PR adds a new Sass variable (`$vr-border-width`) to customize the width of the vertical rule helper.

### Motivation & Context

Vertical rule helper should offer the same customization options as horizontal rule helper.

IMO, this new feature, if landed, should target v5.4.0.

### Type of changes

- [x] New feature (non-breaking change which adds functionality)

### Checklist

- [x] I have read the [contributing guidelines](https://github.com/twbs/bootstrap/blob/main/.github/CONTRIBUTING.md)
- [x] My code follows the code style of the project _(using `npm run lint`)_
- [x] My change introduces changes to the documentation
- [x] I have updated the documentation accordingly
- (N/A) I have added tests to cover my changes
- [x] All new and existing tests passed

#### Live previews

- https://deploy-preview-38886--twbs-bootstrap.netlify.app/docs/5.3/helpers/vertical-rule/

### Related issues

Closes #38856

### CODE REVIEW

File: scss/_variables.scss
Comment (simo026q) [inline scss/_variables.scss:1]: How about added a new variable called `$rule-border-width`?
```scss
$rule-border-width: var(--#{$prefix}border-width) !default;

$hr-border-width: $rule-border-width !default;
$vr-border-width: $rule-border-width !default;
```

You could also do the same with the `$hr-opacity` since you most likely don't know it's used by the vertical rule without looking at the implementation. But that should maybe be another PR

**Review by mdo — APPROVED** [review]

**Comment by mdo** [comment]
Oops, meant to merge another branch. Oh well, this is probably fine to be in v5.3.1 anyway even though technically it's a new Sass variable.
